### ISSUE TITLE
the moving_mean and moving_variance in batchnormalization going to very big after 20 epoch 

### ISSUE THREAD

**Issue by jackyesf** [tensorflow/tensorflow#44371]
tensorFlow version (use command below):  tensorflow 2.2
Python version: 3.7.9
CUDA/cuDNN version: 10.1

 I'm traing a resnet50 model,but the moving_mean and moving_variance in batchnormalization going to very big after 20 epoch ,and when i use this model to predit the mean and variance of the same layer is different with the moving_mean and moving_variance.

bn_conv1 is the first batchnormalization. bn5c_branch2b (BatchNormalization) in the last identity_block:
bn_conv1 (BatchNormalization)
![1](https://user-images.githubusercontent.com/52857358/97404807-a22ecc80-1931-11eb-9c06-f3a288cf8d04.png)
bn5c_branch2b (BatchNormalization):
![2](https://user-images.githubusercontent.com/52857358/97404847-b2df4280-1931-11eb-9a74-8f8ecc2cae6d.png)

               
__________________________________________________________________________________________________
conv1 (Conv2D)                  (None, 402, 6, 32)   224         reshape[0][0]                    
__________________________________________________________________________________________________
bn_conv1 (BatchNormalization)   (None, 402, 6, 32)   128         conv1[0][0]                      
__________________________________________________________________________________________________
leaky_re_lu (LeakyReLU)         (None, 402, 6, 32)   0           bn_conv1[0][0]                   
__________________________________________________________________________________________________
max_pooling2d (MaxPooling2D)    (None, 200, 6, 32)   0           leaky_re_lu[0][0]                
........
........

res5c_branch2b (Conv2D)         (None, 13, 6, 128)   147584      leaky_re_lu_46[0][0]             
__________________________________________________________________________________________________
bn5c_branch2b (BatchNormalizati (None, 13, 6, 128)   512         res5c_branch2b[0][0]             
__________________________________________________________________________________________________
leaky_re_lu_47 (LeakyReLU)      (None, 13, 6, 128)   0           bn5c_branch2b[0][0]              
__________________________________________________________________________________________________
res5c_branch2c (Conv2D)         (None, 13, 6, 1024)  132096      leaky_re_lu_47[0][0]             
__________________________________________________________________________________________________
bn5c_branch2c (BatchNormalizati (None, 13, 6, 1024)  4096        res5c_branch2c[0][0]             
__________________________________________________________________________________________________
add_15 (Add)                    (None, 13, 6, 1024)  0           bn5c_branch2c[0][0]              
                                                                 leaky_re_lu_45[0][0]             
__________________________________________________________________________________________________
leaky_re_lu_48 (LeakyReLU)      (None, 13, 6, 1024)  0           add_15[0][0]                     
__________________________________________________________________________________________________
max_pooling2d_1 (MaxPooling2D)  (None, 4, 2, 1024)   0           leaky_re_lu_48[0][0]             
__________________________________________________________________________________________________
flatten (Flatten)               (None, 8192)         0           max_pooling2d_1[0][0]            
__________________________________________________________________________________________________
output_Dense (Dense)            (None, 1)            8193        flatten[0][0]                    
===================================================================================

**Comment by ravikyram** [comment]
@jackyesf 

Can you please share colab link or simple standalone code with supporting files to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by jackyesf** [comment]
I tried on minst ,no problem ,but got this when i  test on my dataset .

**Comment by ravikyram** [comment]
@jackyesf 

Will it be possible to share sample of your dataset to test from our side.Also,please share complete code snippet to reproduce the issue. It helps us in localizing the issue faster.Thanks!

**Comment by jackyesf** [comment]
@ravikyram the dataset is too big to upload.

**Comment by jackyesf** [comment]
I have traced bn_conv1 (the first Bacthnormalization layer)  on first few step .the output of this layer make me crazy.
I set the breakpoint in def _fused_batch_norm(self, inputs, training):  
![output of bnconv1](https://user-images.githubusercontent.com/52857358/97807907-79466880-1c9e-11eb-9616-f878bfb298e0.png)


If we use the mini-batch mean and variance to calculate the output is close to the real output catched by pdb.
but i think should use the self.moving_mean and self.moving_variance to calculate the output of bn_conv1 .
    def _fused_batch_norm_training():
      return nn.fused_batch_norm(
          inputs,
          gamma,
          beta,
          mean=self.moving_mean,
          variance=_maybe_add_or_remove_bessels_correction(
              self.moving_variance, remove=False),
          epsilon=self.epsilon,
          is_training=True,
          data_format=self._data_format,
          exponential_avg_factor=exponential_avg_factor)

def fused_batch_norm():    nn_impl.py   line:1426

def fused_batch_norm_v3():     gen_nn_ops.py  line:4200

def fused_batch_norm_v3_eager_fallback():     gen_nn_ops.py  line:4296

def quick_execute():     execute.py

then i can't trace anymore, it's c++

**Comment by jackyesf** [comment]
The batch_size=75

**Comment by jackyesf** [comment]
def identity_block(X,f,filters,stage,block):
    conv_name_base = "res"+str(stage)+block+"_branch"
    bn_name_base = "bn"+str(stage)+block+"_branch"
    F1,F2,F3 = filters
    X_shortcut = X

    X = Conv2D(filters=F1,kernel_size=(1,1),strides=(1,1),padding="valid",name=conv_name_base+"2a",
               kernel_initializer=VarianceScaling(seed=0),kernel_regularizer=regularizers.l2(weight_decay))(X)
    X = BatchNormalization(axis=-1,name=bn_name_base+"2a")(X,training=True)
    X = Activation("relu")(X)
    
    X = Conv2D(filters=F2,kernel_size=(f,f),strides=(1,1),padding="same",name=conv_name_base+"2b",
               kernel_initializer=VarianceScaling(seed=0),kernel_regularizer=regularizers.l2(weight_decay))(X)
    X = BatchNormalization(axis=-1,name=bn_name_base+"2b")(X,training=True)
    X = Activation("relu")(X)

    X = Conv2D(filters=F3,kernel_size=(1,1),strides=(1,1),padding="valid",name=conv_name_base+"2c",
               kernel_initializer=VarianceScaling(seed=0),kernel_regularizer=regularizers.l2(weight_decay))(X)
    X = BatchNormalization(axis=-1,name=bn_name_base+"2c")(X,training=True)

    X = layers.add([X,X_shortcut])
    X = Activation("relu")(X)
    
    return X

def convolutional_block(X,f,filters,stage,block,s=2):
    conv_name_base = "res" + str(stage) + block + "_branch"
    bn_name_base = "bn" + str(stage) + block + "_branch"
    F1, F2, F3 = filters
    X_shortcut = X

    X = Conv2D(filters=F1, kernel_size=(1, 1), strides=(s, 1), padding="valid",name=conv_name_base + "2a", 
               kernel_initializer=VarianceScaling(seed=0),kernel_regularizer=regularizers.l2(weight_decay))(X)
    X = BatchNormalization(axis=-1,name=bn_name_base + "2a")(X,training=True)
    X = Activation("relu")(X)

    X = Conv2D(filters=F2, kernel_size=(f, f), strides=(1, 1), padding="same",name=conv_name_base + "2b", 
               kernel_initializer=VarianceScaling(seed=0),kernel_regularizer=regularizers.l2(weight_decay))(X)
    X = BatchNormalization(axis=-1,name=bn_name_base + "2b")(X,training=True) 
    X = Activation("relu")(X)

    X = Conv2D(filters=F3, kernel_size=(1, 1), strides=(1, 1), padding="valid",name=conv_name_base + "2c", 
               kernel_initializer=VarianceScaling(seed=0),kernel_regularizer=regularizers.l2(weight_decay))(X)
    X = BatchNormalization(axis=-1,name=bn_name_base + "2c")(X,training=True)

    X_shortcut = Conv2D(filters=F3,kernel_size=(1,1),strides=(s,1),padding="valid",name=conv_name_base+"1",
                        kernel_initializer=VarianceScaling(seed=0),kernel_regularizer=regularizers.l2(weight_decay))(X_shortcut)
    X_shortcut = BatchNormalization(axis=-1,name=bn_name_base+"1")(X_shortcut,training=True)
 
    X = layers.add([X, X_shortcut])
    X = Activation("relu")(X)
    return X

input_tensor = Input(shape=(date_days,6))
input_tensor_reshape = keras.layers.Reshape((date_days,6,1))(input_tensor)

Resnet_X = Conv2D(32,kernel_size=(3,2),strides=(3,1),name="conv1",kernel_initializer=VarianceScaling(seed=0),padding="same",
                  kernel_regularizer=regularizers.l2(weight_decay))(input_tensor_reshape)
Resnet_X = BatchNormalization(axis=-1,name="bn_conv1")(Resnet_X,training=True)
Resnet_X = Activation("relu")(Resnet_X) 
Resnet_X = AveragePooling2D(pool_size=(3,1),strides=(2,1))(Resnet_X)

    # Stage 2
Resnet_X = convolutional_block(Resnet_X,f=3,filters=[32,32,128],stage=2,block="a",s=2)
Resnet_X = identity_block(Resnet_X,f=3,filters=[32,32,128],stage=2,block="b")
Resnet_X = identity_block(Resnet_X,f=3,filters=[32,32,128],stage=2,block="c")

    #Stage 3
Resnet_X = convolutional_block(Resnet_X,f=3,filters=[64,64,256],stage=3,block="a",s=2)
Resnet_X = identity_block(Resnet_X,f=3,filters=[64,64,256],stage=3,block="b")
Resnet_X = identity_block(Resnet_X,f=3,filters=[64,64,256],stage=3,block="c")
Resnet_X = identity_block(Resnet_X,f=3,filters=[64,64,256],stage=3,block="d")

    # Stage 4
Resnet_X = convolutional_block(Resnet_X,f=3,filters=[128,128,512],stage=4,block="a",s=2)
Resnet_X = identity_block(Resnet_X,f=3,filters=[128,128,512],stage=4,block="b")
Resnet_X = identity_block(Resnet_X,f=3,filters=[128,128,512],stage=4,block="c")
Resnet_X = identity_block(Resnet_X,f=3,filters=[128,128,512],stage=4,block="d")
Resnet_X = identity_block(Resnet_X,f=3,filters=[128,128,512],stage=4,block="e")
Resnet_X = identity_block(Resnet_X,f=3,filters=[128,128,512],stage=4,block="f")

    #Stage 5
Resnet_X = convolutional_block(Resnet_X,f=3,filters=[256,256,1024],stage=5,block="a",s=2)
Resnet_X = identity_block(Resnet_X,f=3,filters=[128,128,1024],stage=5,block="b")
Resnet_X = identity_block(Resnet_X,f=3,filters=[128,128,1024],stage=5,block="c")
    
Resnet_X = MaxPooling2D(pool_size=(3,3))(Resnet_X)
    
Resnet_X = Flatten()(Resnet_X)   
output_tensor = Dense(1,activation="sigmoid",name="output_Dense",kernel_initializer=VarianceScaling(seed=0))(Resnet_X)
model = Model(input_tensor,output_tensor,dynamic = True)


opt = tf.keras.optimizers.Nadam(learning_rate=0.0007)

lossf = weighted_cross_entropy(8.26)
model.compile(optimizer=opt,loss=[lossf],metrics=['accuracy',tf.keras.metrics.Precision(),tf.keras.metrics.Recall(),tf.keras.metrics.AUC(),F1])
